File: DDUtils-IOS/ui/DDTextField [ios]/DDTextFieldDemo_ios/ViewController.m
//
//  ViewController.m
//  DDTextFieldDemo_ios
//
//  Created by Dominik Pich on 09/02/15.
//  Copyright (c) 2015 Dominik Pich. All rights reserved.
//

#import "ViewController.h"
#import "DDTextField.h"

@interface ViewController () <DDTextFieldDelegate>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (BOOL)textFieldShouldReturn:(DDTextField *)textField {
    return textField.isMultiLine;
}

@end

File: swift/ddutils-common/model/UserDefaults+RunOnce [ios + osx + demo]/RunOnceDemo/ViewController.swift
//
//  ViewController.swift
//  RunOnceDemo
//
//  Created by Dominik Pich on 10/12/16.
//  Copyright © 2016 Dominik Pich. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        
        let run = UserDefaults.standard.runOnce(key: "first") { () -> Bool in
            self.view.backgroundColor = UIColor.red
            print("first run ONLY")
            
            return false //dont run again
        }
        
        if(!run) {
            //all but first
            self.view.backgroundColor = UIColor.green
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

